### PR TITLE
Everything is cmake

### PR DESCRIPTION
**Pull request by Electroid**
### What does this PR do?

This is a work-in-progress PR that will consolidate all of our build scripts to `cmake`. The goal is that you only need to run `cmake` to build Bun from scratch.

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [ ] Code changes

### CODE REVIEW

**Comment by robobun** [comment]
@Electroid, your commit 2e26b66f9ef8362b7c702155564d38d406618060 is building: [`#2102`](https://buildkite.com/bun/bun/builds/2102)

File: cmake/tools/SetupWebKit.cmake [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+file(REMOVE ${CACHE_PATH}/${WEBKIT_FILENAME})
+file(REMOVE_RECURSE ${WEBKIT_PATH})
+file(RENAME ${CACHE_PATH}/bun-webkit ${WEBKIT_PATH})
+
+if(APPLE)
+  file(REMOVE_RECURSE ${WEBKIT_INCLUDE_PATH}/unicode)
+endif()
+
+# --- WebKit ---
+# WebKit is either prebuilt and distributed via NPM, or you can pass WEBKIT_PATH to use a local build.
```
Comment (Jarred-Sumner) [inline cmake/tools/SetupWebKit.cmake:83]: we don't distribute via npm

Comment (Jarred-Sumner) [reply cmake/tools/SetupWebKit.cmake:83]: its too large

File: cmake/analysis/RunIWYU.cmake [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# IWYU = "Include What You Use"
+# https://include-what-you-use.org/
```
Comment (Jarred-Sumner) [inline cmake/analysis/RunIWYU.cmake:2]: are we using this?

File: cmake/analysis/RunCppLint.cmake [hunk 3]
```diff
@@ -0,0 +1,24 @@
+include(Macros)
```
Comment (Jarred-Sumner) [inline cmake/analysis/RunCppLint.cmake:1]: we aren't using cpp lint right? just clang-tidy

cpplint looked deprecated when i last checked

**Pull request merged**